source/_posts/2015-07-06-when-is-ssh-a-insecure.markdown

```diff
@@ -0,0 +1,183 @@
+
+## Experiment
+
+To follow this experiment you'll need to have [vagrant](https://www.vagrantup.com/) installed
+in your computer. You don't need to have previous knowledge of vagrant
+to follow the experiment.
+All the config files used in this experiment can be found [here](https://github.com/marpo60/ssh_forward_agent_issue).
+
+Let's start building a topology from scratch.
+
+The topology will include 3 virtual servers and our own machine:
+
+- host
+- web (172.17.8.101)
+- worker (172.17.8.102)
+- secret (172.17.8.110)
+
+Users:
+
+- me
+- Alice (Good)
+- Eve (Bad)
+
+Facts:
+
+* The host machine is our machine where we will create the network
+topology using virtual machines
+* The web server is a virtual machine that will emulate a server
+accessible from the internet.
+* The worker server is a virtual machine that will emulate a
+server accessible only from the web server.
+* The secret server is a virtual machine that will emulate a
+server accessible only by Alice from the internet.
+
+Let's start the servers with `vagrant up`.
+
+`host$ vagrant up` (This can take a while...)
+
+First let's add all the keys to the ssh agent of the host
+machine
+
+{% codeblock lang:sh %}
+  me@host$ ssh-add alice_id_rsa
+  me@host$ ssh-add eve_id_rsa
+{% endcodeblock %}
+
+Now we'll open a new terminal and emulate that Alice is working
+on the worker server
+
+Note that for this experiment we set up another instance
+of the ssh daemon in port 2222
+in order to avoid messing with the default ssh server and
+Vagrant configurations, please check the config files.
+
+{% codeblock lang:sh %}
+  me@host$ ssh -A -i alice_id_rsa -p 2222 alice@172.17.8.101
+  alice@web:~$ ssh -p 2222 172.17.8.102
+  alice@worker:~$
+{% endcodeblock %}
+
+Now on other terminal let's pretend to be Eve
+
+{% codeblock lang:sh %}
+  me@host$ ssh -i eve_id_rsa -p 2222 eve@172.17.8.101
+{% endcodeblock %}
+
+Eve will not use `-A` options because she doesn't need to use the
+worker server
+
+Eve finds out that alice is connected
+
+{% codeblock lang:sh %}
+  eve@web:~$ ps -aux | grep sshd | grep alice
+  root      2469  0.0  0.8 107696  4252 ?        Ss   21:49   0:00
+  sshd: alice [priv]
+  alice     2522  0.0  0.3 107696  1952 ?        S    21:49   0:00
+  sshd: alice@pts/1
+  eve@web:~$
+{% endcodeblock %}
+
+And she also finds out that Alice is using `-A` because a socket
+file was generated in /tmp/ssh-\*
+
+{% codeblock lang:sh %}
+  eve@web:~$ ls -lha /tmp/
+  total 24K
+  drwxrwxrwt  5 root    root    4.0K Jan 19 21:55 .
+  drwxr-xr-x 23 root    root    4.0K Jan 19 21:34 ..
+  drwxrwxrwt  2 root    root    4.0K Jan 19 21:34 .ICE-unix
+  drwx------  2 alice   alice   4.0K Jan 19 21:49
+  ssh-xeIzefGEYs
+  -rwx--x--x  1 vagrant vagrant  676 Jan 19 21:34 vagrant-shell
+  drwxrwxrwt  2 root    root    4.0K Jan 19 21:34 .X11-unix
+  eve@web:~$
+{% endcodeblock %}
+
+Eve now becomes root and impersonate into Alice.
+
+{% codeblock lang:sh %}
+  eve@web:~$ sudo su
```

Comment:
What about adding a note saying that the eve's password is `eve123`?
I had to look at the bootstrap script to figure it out.